service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/C21OrderControllerTest.java

```diff
@@ -0,0 +1,213 @@
+            CallbackRequest.class);
+
+        CaseData caseData = mapper.convertValue(caseDetails.getCaseDetails().getData(), CaseData.class);
+
+        Map<String, Object> templateData = createTemplatePlaceholders();
+
+        given(createC21OrderService.getC21OrderTemplateData(caseData))
+            .willReturn(templateData);
+        given(docmosisDocumentGeneratorService.generateDocmosisDocument(templateData, C21))
+            .willReturn(docmosisDocument);
+        given(uploadDocumentService.uploadPDF(USER_ID, AUTH_TOKEN, pdf, C21_DOCUMENT_TITLE + "_1.pdf"))
+            .willReturn(document);
+
+        MvcResult response = makeRequest(caseDetails, "mid-event");
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = mapper.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        CaseData responseCaseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        C21Order c21Order = responseCaseData.getTemporaryC21Order();
+
+        assertThat(c21Order.getC21OrderDocument().getUrl()).isEqualTo(document.links.self.href);
+        assertThat(c21Order.getC21OrderDocument().getFilename()).isEqualTo(document.originalDocumentName);
+        assertThat(c21Order.getC21OrderDocument().getBinaryUrl()).isEqualTo(document.links.binary.href);
+    }
+
+    @Test
+    void aboutToSubmitShouldPopulateC21BundleAndRemoveTemporariesFromCaseData() throws Exception {
+        byte[] pdf = {1, 2, 3, 4, 5};
+        DocmosisDocument docmosisDocument = new DocmosisDocument(C21_DOCUMENT_TITLE, pdf);
+        Document document = document();
+
+        CallbackRequest caseDetails = mapper.readValue(readBytes("core-case-data-store-api/callback-request.json"),
+            CallbackRequest.class);
+
+        CaseData caseData = mapper.convertValue(caseDetails.getCaseDetails().getData(), CaseData.class);
+
+        Map<String, Object> templateData = createTemplatePlaceholders();
+
+        given(createC21OrderService.getC21OrderTemplateData(caseData))
+            .willReturn(templateData);
+        given(docmosisDocumentGeneratorService.generateDocmosisDocument(templateData, C21))
+            .willReturn(docmosisDocument);
+        given(uploadDocumentService.uploadPDF(USER_ID, AUTH_TOKEN, pdf, C21_DOCUMENT_TITLE + "_1.pdf"))
+            .willReturn(document);
+
+        MvcResult response = makeRequest(caseDetails, "about-to-submit");
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = mapper.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        CaseData responseCaseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        assertThat(responseCaseData.getTemporaryC21Order()).isEqualTo(null);
+        assertThat(responseCaseData.getJudgeAndLegalAdvisor()).isEqualTo(null);
+
```

Comment:
I would just add another assertion here that responseCaseData.getC210OrderBundle().isSize(1) or something 🤷‍♂ you test title reads populate bundle (size(1)) and remove old stuff, the two null assertions